config/config.go

```diff
@@ -281,9 +281,24 @@ func (cfg *Config) NewNode() (host.Host, error) {
 		return nil, err
 	}
 
+	addrsFactory := bhost.DefaultAddrsFactory
+	if cfg.AddrsFactory != nil {
+		addrsFactory = cfg.AddrsFactory
+	}
```

Comment:
Now this is a big mess (only slightly bigger than it was before). I suggest fixing this when we fix our address handling.